### DISCUSSION TITLE
Dionaea binary in Kibana

### DISCUSSION THREAD

**Discussion by V0lundr** [telekom-security/tpotce#1199] [Q&A] [answered]
Hi, When a certain binary gets dropped under /data/dionaea/binaries, is there a way to trace it in the Kibana logs to understand it's origin and the context around it, like with Cowrie binary/hash/logs for example?

**Comment by t3chn0m4g3** [comment]
No, there is no info available in Kibana since Dionaea does not log this information.

**Comment by vertebarbe** [comment]
Hello, I had the same problem, so I forked dionaea's 0.11.0 version here: https://github.com/vertebarbe/dionaea
I added logging of md5 and sha256 hashes when a file is uploaded.

@V0lundr, if you want to add it to T-Pot you can:
1. checkout the relevant commit here: https://github.com/vertebarbe/dionaea/commit/dae46764cae8947453338fca71e51842101e2994
2. copy the files log_json.py and store.py on your T-Pot instance
3. Add the following lines to your /opt/tpot/etc/tpot.yml in the "volumes" section of the dionaea service
\- /path/to/store.py:/opt/dionaea/lib/dionaea/python/dionaea/store.py
\- /path/to/log_json.py:/opt/dionaea/lib/dionaea/python/dionaea/log_json.py

If you are interested, I also added some support for basic smb and ftp uploads in the next commits.

**Reply by Rocco-Hash1** [reply]
Hey @vertebarbe , have you been able to get the binaries out of data/dionaea/binaries to the local machine? Is so how did you do it? I tried scp -p 64295 filelocation destinationlocation but i get permission denied after changing permissions on the directory any help greatly appreciated

**Reply by t3chn0m4g3** [reply]
On T-Pot add the user you want to `scp` with to the `tpot`-group: `sudo usermod -aG tpot <scp user>`.

**Reply by Rocco-Hash1** [reply]
@t3chn0m4g3 thanks but when I run sudo usermod -aG tpot john i get user does not exist and all the man pages for usermod say that is how you add a user to the group, any other ideas? also i have tried different variations of  sudo usermod -a -G tpot john command same response

**Reply by vertebarbe** [reply]
Hi @Rocco-Hash1, to make the scp command work, you need to use the following command:
`scp <remote_username>@<tpot_ip>:/path/to/file /where/to/put` where `remote_username` is the user you normally use to remotely connect to the machine (if not sure, execute `whoami` **from inside the T-Pot instance**).

An alternative is to **create a new user on the T-Pot instance** and add it to the `tpot` group as mentioned before, and use this user as `remote_username` in the scp command.

If you still have permission issues, make sure that you have enough permissions **on your local machine**. You can try to `scp` into the `/tmp` folder on Linux/MacOS, or `%temp%` on Windows, which should always be writable.

If you have further issues, please put the exact command(s) executed and error(s). You can also attach a screenshot.
**Don't forget to redact sensitive information like ip/username/domain**.

**Reply by Rocco-Hash1** [reply]
@vertebarbe wait am i supposed to execute the scp command on tpot? i have been executing on the kali vm i am using to connect to the tpot running in MS Azure? i was executing this command on kali vm attempting to connect to tpot in azure scp -P 64295 tec@ip.ip.ip.ip:/data/dionaea/binaries/ae12bb54af31227017feffd9598a6f5e . permission denied is what i get but from what you saying sounds like I should be executing command on tpot?

**Reply by vertebarbe** [reply]
No, you need to execute the command on your Kali vm, but using the username of your T-Pot instance. From your command it seems you are using user `tec` instead of `tsec`, please try with `tsec`

**Reply by Rocco-Hash1** [reply]
yeah thats a typo on my part im sorry i was using tsec so that should work right? because its not and neither is adding the user john from my kali vm to tpot via sudo usermod -a -G tpot john gives me user doesnt exist so i am all out of ideas any help greatly appreciated if you can think of something, someone mentioned installing vnc on tpot what do you think?

**Reply by vertebarbe** [reply]
The `user does not exist` error is normal because there is no user `john` on tpot by default, but you should not need it anyway. Are you able to connect to the tpot instance via ssh?

**Reply by vertebarbe** [reply]
What you can try is use scp with the root user you setup when creating the tpot instance in azure

**Reply by Rocco-Hash1** [reply]
yeah if i use ssh -P 64295 tsec@ip.ip.ip.ip i get prompted for password and I log in that how i found where the binaries were located

**Reply by vertebarbe** [reply]
Also please try the following:
`scp -P 64295 tsec@ip.ip.ip.ip:/data/dionaea/binaries/ae12bb54af31227017feffd9598a6f5e /tmp` (scp into `/tmp` instead of current folder)

**Reply by Rocco-Hash1** [reply]
@vertebarbe thank you hear is a screenshot, sorry took so long attackers took out my tpot. see what its saying and when i change name to tsec and try to use the password i have set for the machine it doesnt work, any help greatly appreciated
![honeypot](https://github.com/telekom-security/tpotce/assets/89274294/92db24d0-32dc-4d44-b176-7e73cc960593)

**Reply by vertebarbe** [reply]
Ok so you can try to add the user `tblack` to the tpot group by running this command **on tpot**:
`sudo usermod -aG tpot tblack`
and then running the last scp command again

**Reply by Rocco-hash** [reply]
hey @vertebarbe thanks for your help so far so it appears to be an issue with the permissions associated with the /data/dionaea/binaries directory look at following screenshots for evidence i can download the files from cowrie but not dionaea any ideas so i can get the binaries from dionaea greatly appreciated
![honeypot1](https://github.com/telekom-security/tpotce/assets/64816090/96393f37-7225-4714-bd79-45fe8f0854cd)
![honeypot2](https://github.com/telekom-security/tpotce/assets/64816090/0613298e-25aa-4292-a0b5-ffbde60eff57)

**Reply by Rocco-Hash1** [reply]
@vertebarbe you got any ideas ?

**Reply by vertebarbe** [reply]
Did you manually change the permissions of the Dionaea directory? If so, then make sure to put them back to default by comparing to the cowrie folder, otherwise I don't have any idea of what it could be :/

**Reply by Rocco-Hash1** [reply]
@vertebarbe no i did not manually change the permissions associated with the directory all i did was sudo usermod -a -G tpot tblack tried to download the binaries form Dionaea and then decided I would try cowrie and it worked went back to Dionaea as the output shows @t3chn0m4g3 do you have any ideas? @vertebarbe thank you so much for your help

**Reply by Rocco-Hash1** [reply]
@vertebarbe here is screenshot

![tpot32](https://github.com/telekom-security/tpotce/assets/89274294/df22999f-2888-43b2-a8c9-1beb9bd50226)

**Reply by Rocco-Hash1** [reply]
see the binaries are not even showing in the directory which is also odd when i run ls they show up

**Reply by vertebarbe** [reply]
Ok I think I see your issue, so the thing is that every day and possibly every restart there is something called `logrotate` that archives the downloaded files to avoid it taking too much space. You should be able to see them in the `/data/dionaea` folder (should be `binaries.tgz.<number>`). You can try listing the archive contents with `tar -tvf <filename>` and see if there is the file you were looking for in one of them.

**Reply by Rocco-Hash1** [reply]
@vertebarbe thanks for showing me that so cool, but i cant get the binaries out 

![tpot2](https://github.com/telekom-security/tpotce/assets/89274294/2defdc72-7694-4ef3-a34e-00860ae9f717)

**Reply by Rocco-Hash1** [reply]
@vertebarbe you got any ideas, is this a known issue with tpot? should i file an issue?

**Reply by vertebarbe** [reply]
I don't have any idea of why this would be happening, sorry, you can try creating an issue about this.

**Reply by t3chn0m4g3** [reply, marked as answer]
This is not a T-Pot issue and usually this does the trick ...

> On T-Pot add the user you want to `scp` with to the `tpot`-group: `sudo usermod -aG tpot <scp user>`.

If this does not work, it is a typical Linux permission issue, nothing we can fix.

**Reply by Rocco-Hash1** [reply]
all good @vertebarbe thank you so much if it wasnt for you I wouldnt have gotten this far, thank you!! I'm going to take tpot offline until i can get an answer going to file an issue thanks again

**Reply by Rocco-Hash1** [reply]
> This is not a T-Pot issue and usually this does the trick ...
> 
> > On T-Pot add the user you want to `scp` with to the `tpot`-group: `sudo usermod -aG tpot <scp user>`.
> 
> If this does not work, it is a typical Linux permission issue, nothing we can fix.

so what would you suggest the screenshots show its not working what would you suggest

**Reply by Rocco-Hash1** [reply]
@t3chn0m4g3 you was right i go it was a permission issue thank you and @vertebarbe for your help

**Comment by Rocco-Hash1** [comment]
thanks @vertebarbe they took out the tpot again so i got to spin up another once it is all done I will reply back i hope it works thank you so much for your assistance so far
